File: Level1/로또의 최고 순위와 최저 순위.py
#로또의 최고 순위와 최저 순위
lottos=[44, 1, 0, 0, 31, 25]
win_nums=[31, 10, 45, 1, 6, 19]
def solution(lottos, win_nums):
    answer = []
    cnt2=0 #최저 순위 체크 변수
    for lotto in lottos:
        if lotto in win_nums: #몇 개 맞나 확인
            cnt2+=1
    cnt1=cnt2+lottos.count(0) #최고 순위는 최저+'0'의 갯수
    if cnt1==6:
        answer.append(1)
    elif cnt1==5:
        answer.append(2)
    elif cnt1==4:
        answer.append(3)
    elif cnt1 == 3:
        answer.append(4)
    elif cnt1==2:
        answer.append(5)
    elif cnt1<=1:
        answer.append(6)
    if cnt2==6:
        answer.append(1)
    elif cnt2==5:
        answer.append(2)
    elif cnt2==4:
        answer.append(3)
    elif cnt2 == 3:
        answer.append(4)
    elif cnt2==2:
        answer.append(5)
    elif cnt2<=1:
        answer.append(6)
    return answer
print(solution(lottos,win_nums))

File: Level1/로또의 최고 순위와 최저순위.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

//0의 개수 반환
int get_zero_count(vector<int>& lottos) {
    int cnt = 0;
    for (int i = 0; i < lottos.size(); ++i)
        if (lottos[i] == 0)
            cnt++;
    return cnt;
}

//등수 반환 함수
int get_rank(int n) {
    //각 경우의 수가 몇 위에 해당하는 지 확인
    
    //1등인 경우
    if (n == 6) 
        return 1;
    //2등인 경우
    else if (n == 5) 
        return 2;
    //3등인 경우
    else if (n == 4) 
        return 3;
    //4등인 경우
    else if (n == 3) 
        return 4;
    //5등인 경우
    else if (n == 2) 
        return 5;
    //6등인 경우
    else 
        return 6;
}

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    //방문리스트
    vector<bool> visited(lottos.size(), false);
    
    //최악의 경우
    int min_cnt = 0;

    //이진탐색을 위한 오름차순정렬
    sort(lottos.begin(), lottos.end());
    
    //몇개 맞는 지 확인
    for (int i = 0; i < lottos.size(); ++i) {
        //현재 로또 번호가 당첨번호에 몇 개있는지 확인
        for (int j = 0; j < win_nums.size(); ++j) {
            if (!visited[i] && lottos[i] == win_nums[j]) {
                min_cnt++;
                //중복할 경우 대비
                visited[i] = true;
            }
        }
    }

    //가장 좋은 경우
    //즉 남아있는 것들도 맞고 지워진 갯수 또한 맞을 경우
    int max_cnt = min_cnt + get_zero_count(lottos);

    answer.push_back(get_rank(max_cnt));
    answer.push_back(get_rank(min_cnt));

    return answer;
}
